### PR TITLE
=act #17206 properly document semantics of context.stop(self)

### PR DESCRIPTION
**Pull request by rkuhn**

### CODE REVIEW

**Comment by rkuhn** [comment]
fixes #17206

**Comment by ktoso** [comment]
LGTM, nice

File: akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala [hunk 1]
```diff
@@ -338,6 +338,9 @@ trait ActorRefFactory {
   /**
    * Stop the actor pointed to by the given [[akka.actor.ActorRef]]; this is
    * an asynchronous operation, i.e. involves a message send.
+   * If this method is applied to the `self` reference from inside an Actor
+   * then that Actor is guaranteed to not process any further messages before
```
Comment (drewhk) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: before terminating?

Comment (rkuhn) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: yes, WDYM?

Comment (drewhk) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: well, will it process further messages after termination?

Comment (rkuhn) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: ah, I see :-) I’d still leave this in to make it absolutely clear that message dequeuing does not only stop after termination but _right now_.

Comment (drewhk) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: then why don't you just say that it is guarenteed to not process any further message *after this call*

Comment (rkuhn) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: yup, better, done that

**Pull request merged**